pkg/operator/ceph/cluster/osd/osd.go

```diff
@@ -734,10 +734,18 @@ func getNodeOrPVCName(d *appsv1.Deployment) (string, error) {
 		return v, nil // OSD is on PVC
 	}
 	for k, v := range d.Spec.Template.Spec.NodeSelector {
-		if k == corev1.LabelHostname {
+		if k == k8sutil.LabelHostname() {
```

Comment:
If we added both these checks in this loop would it remove the need for the new loop below?
```suggestion
		if k == corev1.LabelHostname {
			return v, nil
		}
		if k == k8sutil.LabelHostname() {
			return v, nil
		}
```

Reply:
i think it will work only for use-case where we upgrade from `corev1.LabelHostname`. But we will still need second loop for cases when we moved from one customHostnameLabel to another